doc/sphinx/reference/onedim/nonlinear-solver.md

```diff
@@ -1,5 +1,387 @@
+to assess when the method has reached a point where the iterations can be stopped. This
+point is called convergence. Cantera's implementation uses a **weighted norm** of the
+step vector to determine convergence, rather than a simple absolute norm. A damped
+Newton step is considered to be converged when the **weighted norm** of the correction
+vector is less than 1. During the solution, the process of finding and taking a damped
+Newton step is repeated until the **weighted norm** of the correction vector is less
+than 1, if it is not, then the process continues.
+
+In a multivariate system, different variables may have vastly different magnitudes
+and units. A simple absolute norm could either be dominated by large components or fail
+to account for smaller components effectively. By normalizing the step vector
+components using $w_n$, the weighted norm ensures that the convergence criterion is
+meaningful across all solution components, regardless of their individual scales.
+
+This approach provides a more robust and scale-invariant method for assessing
+convergence, making it especially useful in systems with diverse variables.
+
+#### Definition of the Weighted Norm
+
+The weighted norm of the step vector $\mathbf{s}$ is calculated as:
+
+$$
+\text{Weighted Norm} = \sum_{n,j} \left(\frac{s_{n,j}}{w_n}\right)^2
+$$
+
+where:
+- $s_{n,j}$ is the Newton step vector component for the $n$-th solution variable
+  at the $j$-th grid point.
+- $w_n$ is the error weight for the $n$-th solution component, given by:
+
+$$
+w_n = \epsilon_{r,n} \cdot \frac{\sum_j |x_{n,j}|}{J} + \epsilon_{a,n}
+$$
+
+Here:
+- $\epsilon_{r,n}$ is the relative error tolerance for the $n$-th solution component.
+- $\frac{\sum_j |x_{n,j}|}{J}$ is the average magnitude of the $n$-th solution
+  component over all grid points, and $J$ is the total number of grid points.
+- \( \epsilon_{a,n} \) is the absolute error tolerance for the $n$-th solution
+  component.
+
+#### Interpretation of the Weighted Norm
+
+The weighted norm is a relative measure that helps bring all components of the step
+vector into a comparable range, taking into account the scales of the different
+solution components. It can be interpreted as follows:
+
+- **Relative Error Term** $(\epsilon_{r,n})$: Scales the step size relative to the
+  average magnitude of the corresponding solution component. This means that larger
+  components can tolerate larger steps.
+- **Absolute Error Term** $(\epsilon_{a,n})$: Ensures that even very small solution
+  components are considered in the convergence check by providing a minimum threshold.
+
+#### Convergence Criterion
+
+The Newton iteration is considered converged when the weighted norm is less than 1:
+
+$$
+\sum_{n,j} \left(\frac{s_{n,j}}{w_n}\right)^2 < 1
+$$
+
+This criterion indicates that each component of the step vector $s_{n,j}$ is
+sufficiently small relative to the expected precision (as defined by the weights
+$w_n$).
```

Comment:
Perhaps a more intuitive way of phrasing this is that the change in each component is small compared to the relative and absolute error tolerances on that component.